transformer_engine/jax/csrc/modules.cpp

```diff
@@ -778,29 +779,40 @@ void SelfFusedAttnMax512Forward(cudaStream_t stream, void **buffers, const char
 
     auto cu_seqlens_tensor =
         TensorWrapper(cu_seqlens, std::vector<size_t>{batch + 1}, DType::kInt32);
-    auto rng_state_tensor = TensorWrapper(rng_state, std::vector<size_t>{1}, DType::kInt64);
+
+    // The number of rng_state could be 2 or 4, and the dtype is u32
+    // If there are 4 rng_state, see (0, 1), (2, 3) as 2 u32 rng_state respectively
+    NVTE_CHECK(num_rng_state == 2 || num_rng_state == 4);
+    DType rng_state_type = num_rng_state == 4 ? DType::kInt64 : DType::kInt32;
+    auto dummy_rng_state_tensor = TensorWrapper(nullptr, std::vector<size_t>{2}, rng_state_type);
 
     NVTETensorPack aux_output_tensors;
     nvte_tensor_pack_create(&aux_output_tensors);
 
     TensorWrapper query_workspace_tensor;
 
-    nvte_fused_attn_fwd_qkvpacked(qkv_tensor.data(), bias_tensor.data(), s_tensor.data(),
-                                  o_tensor.data(), &aux_output_tensors, cu_seqlens_tensor.data(),
-                                  rng_state_tensor.data(), q_max_seqlen, descriptor.is_training,
-                                  descriptor.scaling_factor, descriptor.dropout_probability,
-                                  NVTE_QKV_Layout::NVTE_QKV_INTERLEAVED, descriptor.bias_type,
-                                  descriptor.mask_type, query_workspace_tensor.data(), stream);
+    nvte_fused_attn_fwd_qkvpacked(
+        qkv_tensor.data(), bias_tensor.data(), s_tensor.data(), o_tensor.data(),
+        &aux_output_tensors, cu_seqlens_tensor.data(), dummy_rng_state_tensor.data(), q_max_seqlen,
+        descriptor.is_training, descriptor.scaling_factor, descriptor.dropout_probability,
+        NVTE_QKV_Layout::NVTE_QKV_INTERLEAVED, descriptor.bias_type, descriptor.mask_type,
+        query_workspace_tensor.data(), stream);
 
     auto *output_s = reinterpret_cast<Tensor *>(aux_output_tensors.tensors[0]);
     output_s->data.dptr = softmax_aux;
 
-    size_t workspace_size =
+    auto workspace_size =
         query_workspace_tensor.shape().data[0] * typeToSize(query_workspace_tensor.dtype());
+    // fused attn workspace size + casted rng state size
```

Comment:
Why the commented code here and bellow?

Reply:
I will remove them, it should not be here

Reply:
Removed in c8615cf3866fa0266d20739ef221b1ccdb5e45ea